draft-ietf-gnap-core-protocol.md

```diff
@@ -3261,12 +3213,99 @@ the token management URL as valid, since the end result is still the token
+    type of used only for calls the RS during a discovery phase as
+    described in {{rs-request-without-token}}.
+
+## Key Formats {#key-format}
+
+Several different places in GNAP require the presentation of key material 
+by value. Proof of this key material MUST be bound to a request, the nature of which varies with
+the location in the protocol the key is used. For a key used as part of a client instance's initial request
+in {{request-client}}, the key value is the client instance's public key, and
+proof of that key MUST be presented in that request. For a key used as part of an
+access token response in {{response-token-single}}, the proof of that key MUST
+be used when presenting the access token.
+
+A key presented by value MUST be a public key in at least one
+supported format. If the key is sent in multiple
+formats, all the keys MUST be the same. 
+
+proof (string)
+: The form of proof that the client instance will use when
+    presenting the key. The valid values of this field and
+    the processing requirements for each are detailed in 
+    {{binding-keys}}. This field is REQUIRED.
 
-The method the client instance uses to send an access token to the RS depends on the value of the
-"key" and "proof" parameters in [the access token response](#response-token-single).
+jwk (object)
+: Value of the public key as a JSON Web Key. MUST
+            contain an "alg" field which is used to validate the signature.
+            MUST contain the "kid" field to identify the key in the signed
+            object.
+
+cert (string)
+: PEM serialized value of the certificate used to
+            sign the request, with optional internal whitespace.
 
-If the key value is the boolean `false`, the access token is a bearer token
+cert#S256 (string)
+: The certificate thumbprint calculated as
+            per [OAuth-MTLS](#RFC8705) in base64 URL
+            encoding.
+
+Additional key types are defined in [a registry TBD](#IANA).
+
+This non-normative example shows a single key presented in multiple
+formats using a single proofing mechanism.
+
+~~~
+    "key": {
+        "proof": "jwsd",
+        "jwk": {
+                    "kty": "RSA",
+                    "e": "AQAB",
+                    "kid": "xyz-1",
+                    "alg": "RS256",
+                    "n": "kOB5rR4Jv0GMeLaY6_It_r3ORwdf8ci_JtffXyaSx8xY..."
+        },
+        "cert": "MIIEHDCCAwSgAwIBAgIBATANBgkqhkiG9w0BAQsFA..."
+    }
+~~~
+
+### Key References {#key-reference}
+
+Keys in GNAP can also be passed by reference such that the party receiving
+the reference will be able to determine the appropriate keying material for
+use in that part of the protocol. 
+
+~~~
+    "key": "S-P4XJQ_RYJCRTSU1.63N3E"
+~~~
+
+Keys referenced in this manner MAY be shared keys. The key reference
+MUST NOT contain any unencrypted private or shared key information.
+
+The means of dereferencing this value are out of scope for this specification.
```

Comment:
I disagree: when used between the Client and AS this is clearly required for interop. Feel free to open an issue.

Reply:
I disagree that we need to specify this method as it's up to the AS to know how to interpret the value and likely up to the AS to issue such a value to the client. So long as the AS knows which key the client instance means here it should be fine. Opened up a new issue for it #154